clang/lib/Frontend/FrontendActions.cpp

```diff
@@ -1137,6 +1137,7 @@ void GetDependenciesByModuleNameAction::ExecuteAction() {
   Preprocessor &PP = CI.getPreprocessor();
   SourceManager &SM = PP.getSourceManager();
   FileID MainFileID = SM.getMainFileID();
+  PP.EnterSourceFile(MainFileID, nullptr, SourceLocation());
```

Comment:
@jansvoboda11 does this look right to you?

Reply:
It does, but it would be nice to know what exactly this enables/fixes. @cachemeifyoucan

Reply:
In order to create an include tree CAS object, it needs to enter a source file so it has an include stack. Without this, it is going to assertion `assert(IncludeStack.size() == 1);` in `IncludeTreeBuilder::finishIncludeTree` and can't produce a MainIncludeTree.

With module name, the main file is just empty. Just call preprocessor enter source file to get an empty TU include tree.

Reply:
Does the main file include tree get used at all? I wonder if we could skip building it for the main file when handling named module.

Reply:
No, it is not used, but main include tree is the parent node for all the module include tree. We still need to fake one.

Reply:
This doesn't seem right to me: we don't need to build this main include-tree at all.  But if it's hard to avoid that then I'm fine with this as a workaround for now.